### DISCUSSION TITLE
[feature request] Option to automatically fetch upstream for forks

### DISCUSSION THREAD

**Discussion by rickstaa** [community/community#16913] [Other Feature Feedback, Questions, & Ideas]
I like the `Fetch upstream` button added to the Web UI. This dramatically simplifies keeping the main branch of a fork up to date! 🚀 

Giving users the ability to auto-sync the main branch of a fork with the upstream would even further increase the user experience. It would provide users that don't develop on the main fork branch, but separate feature branches the ability to keep their fork up to date without checking the interface.

I know that this can be done using [a github-action](https://stackoverflow.com/questions/23793062/can-forks-be-synced-automatically-in-github/61574295#61574295) like [github-repo-sync](https://github.com/marketplace/actions/github-repo-sync), but I think having this ability in the Web UI would be an improvement. Further, I'm not sure if this is technically feasible given the current codebase, but I just wanted to put it out here for the community to review.

### Implementation example

- Add a `keep this fork up to date` checkbox to the `fetch upstream` menu. 
- Add an update `frequency` select element.

**Comment by 3kh0** [comment]
You can just use [`wei/pull`](https://github.com/wei/pull) but yes this would be a nice feature to be added officially.

**Reply by rickstaa** [reply]
@3kh0 Thanks, for bringing that app to my attention! 🚀  It would indeed be nice if this feature could be added to the GitHub interface.
